benchmarks/matmul_benchmark/implementation.py

```diff
@@ -0,0 +1,74 @@
+from common.benchmark_wrapper import BenchmarkWrapper
+import json
+import os
+import torch
+import time
+
+
+class MatmulBenchmark(BenchmarkWrapper): #it's the same class inside benchmark_info.json
+
+    def __init__(self):
+        self.filePath = os.path.dirname(__file__) #os.path.dirname(__file__) returns the path of this file
+
+    def setSettings(self, settings_file):
+        """
+        Open and read the settings file
+        """
+        settings = os.path.join(self.filePath, "settings", settings_file)
+        self.dimension1 = json.load(open(settings, "r"))["dimension1"]
+        self.dimension2 = json.load(open(settings, "r"))["dimension2"]
+
+    def startBenchmark(self, verbosity=None):
+        
+        returnLog = []
+
+        print("The first matrix has dimensions", self.dimension1, "x", self.dimension2)
+        print("The second matrix has dimensions", self.dimension2, "x", self.dimension1)
+
+        start_time = time.time()
+
+        matrix_1 = torch.randn(self.dimension1, self.dimension2, device="cpu")
+        matrix_2 = torch.randn(self.dimension2, self.dimension1, device="cpu")
+
+        matrices_allocation_time = time.time() - start_time
+
+        print("The matrices has been allocated in %s s" % matrices_allocation_time)
+
+        product_cpu_start_time = time.time()
+
+        torch.matmul(matrix_1, matrix_2)
+
+        product_cpu_end_time = time.time() - product_cpu_start_time
+        returnLog.append({"CPU": product_cpu_end_time})
+        
+        print("The product has been performed in %s s using CPU" % product_cpu_end_time)
+
+        # check if GPU is available before to try to use it
+        print("Checking GPU availability...")
+        if torch.cuda.is_available():
+
+            matrix_1 = matrix_1.to("cuda:0")
+            matrix_2 = matrix_2.to("cuda:0")
+            print("GPU is available!")

```

Comment:
In general `device='cuda'` is automatically casted to `device='cuda:0'` on machines with a single GPU available.
I'd just stick to cuda OR read the device ID from the settings!